### ISSUE TITLE
SlicePasef: Algorithm Failure diann.cpp: 37006, DIANN Version : 2.3.0

### ISSUE THREAD

**Issue by DaliJarboui** [vdemichev/DiaNN#1774]
Hi DIANN community,
We are testing and optimizing SlicePasef methods on HeLa digest, and was running DIANN, the methods were created using the [Slice-PASEF - Method Generator Shiny](https://zenodo.org/records/17034921), after the second file DIANN crashed with the error below, was successuful running the first file and crashed in the second files which is a technical replicate of the first one. In the setting I used as detailed below, I tried to speed up the process by changing the speed to ultra fast, and using 1 missed cleavage, Legacy Quantif and MBR deactivated. 
Appreciate it if someone has a suggestion or details on how to make the search, The runs were made on a TimsTOF Ultra2, and data were generated with no issues. I understand that DIANN can handle SlicePASEF methods if not mistaken, I can provide any other details to help troubleshoot the error. (The system used has an AMD threadripper-32cores and 256 GB of RAM)
Thanks for your help.

diann.exe --f "D:\USERS\DALI\SlicePASEF\20251106_06_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_06_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_07_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_07_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_08_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_08_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_11_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_11b_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_12_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_12_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_13_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_13_1ngHela_nT_2-Slice-15Sl-2F-1S-2MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_16_1ngHela_nT_3-Slice-15Sl-2F-2S-4MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_17_1ngHela_nT_3-Slice-15Sl-2F-2S-4MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_17_1ngHela_nT_3-Slice-15Sl-2F-2S-4MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_18_1ngHela_nT_3-Slice-15Sl-2F-2S-4MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_18_1ngHela_nT_3-Slice-15Sl-2F-2S-4MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_21_1ngHela_nT_4-Slice-15Sl-2F-2S-2Rp-8MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_21_1ngHela_nT_4-Slice-15Sl-2F-2S-2Rp-8MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_22_1ngHela_nT_4-Slice-15Sl-2F-2S-2Rp-8MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_22_1ngHela_nT_4-Slice-15Sl-2F-2S-2Rp-8MS1_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_23_1ngHela_nT_4-Slice-15Sl-2F-2S-2Rp-8MS1.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_02_1ngHela_nT_DIA-2R.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_02_1ngHela_nT_DIA-2R_Bo.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_03_1ngHela_nT_DIA-2R.d
" --f "D:\USERS\DALI\SlicePASEF\20251106_03_1ngHela_nT_DIA-2R_Bo.d
" --lib "D:\Databases\Human_Ref_Proteome_102025\report-lib.predicted.speclib" --threads 50 --verbose 1 --out "D:\USERS\DALI\SlicePASEF\DIANN_SlicePASEF\report.parquet" --qvalue 0.01 --matrices  --min-corr 2.0 --time-corr-only --extracted-ms1 --min-cal 500 --min-class 1000 --pre-filter --out-lib "D:\USERS\DALI\SlicePASEF\DIANN_SlicePASEF\report-lib.parquet" --gen-spec-lib --fasta "D:\Databases\Human_Ref_Proteome_102025\Human_Proteome_102025.fasta" --min-pep-len 6 --max-pep-len 50 --min-pr-mz 300 --max-pr-mz 1800 --min-pr-charge 2 --max-pr-charge 4 --min-fr-mz 200 --max-fr-mz 2000 --cut K*,R* --missed-cleavages 1 --unimod4 --var-mods 1 --mass-acc 15 --mass-acc-ms1 15 --individual-mass-acc --individual-windows --peptidoforms --rt-profiling --pg-level 1 --direct-quant  

DIA-NN 2.3.0 Academia  (Data-Independent Acquisition by Neural Networks)
Compiled on Sep 27 2025 22:03:55
Current date and time: Fri Nov  7 21:59:32 2025
CPU: AuthenticAMD AMD Ryzen Threadripper PRO 9975WX 32-Cores
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 SSE4a 
Logical CPU cores: 64
Thread number set to 50
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
Only peak groups with correlation-based score exceeding 2 will be considered; will be deactivated for the second MBR pass
DIA-NN will disable machine learning during the calibration stage to save memory, this may lead to somewhat lower identification numbers; will be deactivated for the second MBR pass
Ultra-fast mode will be used, this may worsen the identification performance; will be deactivated for the second MBR pass
Minimum number of precursors used for calibration set to 500
Minimum number of precursors used for linear classifier training set to 1000
DIA-NN will pre-filter candidate precursors
A spectral library will be generated
Min peptide length set to 6
Max peptide length set to 50
Min precursor m/z set to 300
Max precursor m/z set to 1800
Min precursor charge set to 2
Max precursor charge set to 4
Min fragment m/z set to 200
Max fragment m/z set to 2000
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 1
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Mass accuracy will be determined separately for different runs
Scan windows will be inferred separately for different runs
Peptidoform scoring enabled
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
Implicit protein grouping: protein names; this determines which peptides are considered 'proteotypic' and thus affects protein FDR calculation
Legacy (direct) quantification mode
Mass accuracy will be fixed to 1.5e-05 (MS2) and 1.5e-05 (MS1)

26 files will be processed
[0:00] Loading spectral library D:\Databases\Human_Ref_Proteome_102025\report-lib.predicted.speclib
[0:20] Library annotated with sequence database(s): D:\Databases\Human_Ref_Proteome_102025\Human_Proteome_102025.fasta
[0:24] Spectral library loaded: 20654 protein isoforms, 38354 protein groups and 43212900 precursors in 16009843 elution groups.
[0:24] Loading protein annotations from FASTA D:\Databases\Human_Ref_Proteome_102025\Human_Proteome_102025.fasta
[0:24] Annotating library proteins with information from the FASTA database
[0:24] Gene names missing for some isoforms
[0:24] Library contains 20654 proteins, and 20393 genes
[0:49] Initialising library
WARNING: it is strongly recommended to enable MBR when analysing with a large library, if this is a quantitative analysis

[1:43] File #1/26
[1:43] Loading run D:\USERS\DALI\SlicePASEF\20251106_06_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1.d
[4:18] Analysing MS1 spectra
[4:18] Analysing MS2 spectra
[4:30] Integrating MS1 spectra
[4:30] Integrating MS2 spectra
[4:54] Pre-processing...
[5:12] 11254 MS1 and 11254 MS2 scans in 11254 (inferred) and 11254 (encoded) cycles, 28812950 precursors in range
[5:16] Calibrating with mass accuracies 22 (MS1), 25 (MS2)
[42:30] RT window set to 2.10715
[42:30] IM window set to 0.0384448
[42:30] Peak width: 14.236
[42:30] Scan window radius set to 31
[42:31] Recommended MS1 mass accuracy setting: 14 ppm
[44:57] Searching decoys
[64:30] Main search
[103:52] Removing low confidence identifications
[104:30] Removing interfering precursors
[104:50] Training neural networks on 57059 target and 34530 decoy PSMs
[105:24] Training neural networks on 57059 target and 32956 decoy PSMs
[105:37] IDs at 0.01 FDR: 23663
[105:40] Precursors at 1% peptidoform FDR: 22434
[105:44] Number of IDs at 0.01 FDR: 26169
[105:44] Calculating protein q-values
[105:45] Number of proteins identified at 1% FDR: 3391 (precursor-level), 2919 (protein-level) (inference performed using proteotypic peptides only)
[105:45] Quantification
[105:51] Quantification information saved to D:\USERS\DALI\SlicePASEF\20251106_06_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1.d.quant

[105:52] File #2/26
[105:52] Loading run D:\USERS\DALI\SlicePASEF\20251106_06_1ngHela_nT_1-Slice-15Sl-1F-1S-1MS1_Bo.d
[108:22] Analysing MS1 spectra
[108:22] Analysing MS2 spectra
[108:35] Integrating MS1 spectra
[108:35] Integrating MS2 spectra
[109:00] Pre-processing...
[109:18] 11154 MS1 and 11154 MS2 scans in 11154 (inferred) and 11154 (encoded) cycles, 28812950 precursors in range
[109:22] Calibrating with mass accuracies 22 (MS1), 25 (MS2)
[128:47] RT window set to 2.04711
[128:47] IM window set to 0.0428514
[128:47] Peak width: 13.624
[128:47] Scan window radius set to 29
[128:48] Recommended MS1 mass accuracy setting: 19 ppm
ERROR: algorithmic failure: M:\diann\src\diann.cpp: 37006

DIA-NN exited
diann-stats.exe "D:\USERS\DALI\SlicePASEF\DIANN_SlicePASEF\report.parquet"
PDF report will be generated in the background

**Comment by vdemichev** [comment]
Hi,

Apparently 2.3 has an issue with ultra-fast mode used on timsTOF data. Will be fixed in 2.3.1. Please either (i) don't use the ultra-fast mode or (ii) use 2.2 for now.

Best,
Vadim

**Comment by DaliJarboui** [comment]
Thanks Vadim for the fast answer, appreciate it, will try the version 2.2, 
Any recommendations for the search, was meant just to check the performance of the SlicePasef compared to our standard DIA PASEF, or the setting I used were good enough.? 
Best
Dali
